include/libnuraft/context.hxx

```diff
@@ -88,6 +90,13 @@ public:
         params_ = to;
     }
 
+    global_mgr * get_global_mgr() const {
+        if (custom_global_mgr_ != nullptr) {
+            return custom_global_mgr_;
+        }
+        return nuraft_global_mgr::get_instance();
+    }
+
```

Comment:
`nuraft_global_mgr` shouldn't be involved in `context` and also `context` shouldn't be aware of the singleton. Please move this function out and put it into `raft_server`:
```c++
global_mgr* raft_server::get_global_mgr() const {
    if (ctx_->custom_global_mgr_) {
        return ctx_->custom_global_mgr_;
    }
    return nuraft_global_mgr::get_instance();
}
```